feat(vulkan): Add Vulkan rendering backend implementaion

magicianA opened the pull request:
Implement a Vulkan rendering backend to complement the existing OpenGL and Metal backends. Most drawing actions are supported, with current execution limited to macOS via MoltenVK.

<img width="1280" height="1059" alt="image" src="https://github.com/user-attachments/assets/312036c2-7dd3-4211-91f1-f63b3022118f" />

<img width="1280" height="1059" alt="image" src="https://github.com/user-attachments/assets/96eb58b7-9566-4b2d-bf15-81b4dde68b3c" />


See internal group chat for details.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lynx-family/skity?pullRequest=83) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/lynx-family/skity?pullRequest=83) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lynx-family/skity?pullRequest=83) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lynx-family/skity?pullRequest=83) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/lynx-family/skity?pullRequest=83) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lynx-family/skity?pullRequest=83) it.</sub>

magicianA committed:
feat(vulkan): Add Vulkan rendering backend implementaion

Implement a Vulkan rendering backend to complement the existing OpenGL and
Metal backends. Most drawing actions are supported, with current execution
limited to macOS via MoltenVK.
